cobra/flux_analysis/sampling.py

```diff
@@ -56,6 +56,40 @@
 """
 
 
+def mp_init(obj):
+    """Initialize the multiprocessing pool."""
+    global sampler
+    sampler = obj
+
+
+def shared_np_array(shape, data=None):
+    """Create a new numpy array that resides in shared memory.
+
+    Parameters:
+    ----------
+    shape : tuple of ints
+        The shape of the new array.
+    data : numpy.array
+        Data to copy to the new array. Has to have the same shape.
+    """
+    size = np.prod(shape)
+    array = Array(ctypes.c_double, int(size))
+    np_array = np.frombuffer(array.get_obj())
+    np_array.reshape(shape)
+
+    if data is not None:
+        if len(shape) != len(data.shape):
+            raise ValueError("`data` must have the same dimensions"
+                             "as the created array.")
+        same = all(x == y for x, y in zip(shape, data.shape))
+        if not same:
+            raise ValueError("`data` must have the same shape"
+                             "as the created array.")
+        np_array = data[:]
```

Comment:
No that would be wrong. np_array is the one in shared memory and you have to copy the non-shared values into it.